### ISSUE TITLE
Add Lint and Type check for Pull request

### ISSUE THREAD

**Issue by growupanand** [growupanand/ConvoForm#264]

**Comment by mansi104-ai** [comment]
Please assign this issue to me under gssoc 24

**Comment by growupanand** [comment]
@mansi104-ai assigned to you.

**Comment by growupanand** [comment]
@mansi104-ai , can you please share the status? Let me know if you need any help.

**Comment by mansi104-ai** [comment]
@convoform/api:lint: Pages directory cannot be found at C:\Users\mansi\GSSOC\convoform\ConvoForm\packages\api\pages or C:\Users\mansi\GSSOC\convoform\ConvoForm\packages\api\src\pages. If using a custom path, please configure with the `no-html-link-for-pages` rule in your eslint config file.

I am getting this error on running workflow , could you please help?

I have already tried adding no-html-link-for-pages, but didn't work

**Comment by growupanand** [comment]
@mansi104-ai , Are you still working on this?, because the PR is showing closed.

**Comment by mansi104-ai** [comment]
Yes, sir I am working on it 
Didn't close the PR

**Comment by growupanand** [comment]
@mansi104-ai, can you please share the status?

**Comment by IdrisGit** [comment]
Is this issue still open?

I think you want to do a lint and type check whenever someone opens up a new PR. For that, we can create a new workflow that will run GitHub action whenever someone opens up a new PR against the `main` branch and do the lint and type checks.

Let me know if this is what you want or something else.

I can pick up the issue.

**Comment by growupanand** [comment]
@IdrisGit  yes, you can do this. I want to create a workflow where we check lint and type checks as same which are being done in local machine before doing commit.

This will ensure that if someone skip the pre-commit hook, So still there is not lint or type error in there PR.

**Comment by IdrisGit** [comment]
Ok great, I will create a workflow file that will run on every PR raised.

I am participating in Hacktoberfest this year, can you add hacktoberfest label to this issue?

**Comment by IdrisGit** [comment]
The root `package.json` file has these scripts, that are used in the `pnpm lint && pnpm type-check` command suggested to be used in the issue.

```json
 "lint": "dotenv -e .env.local -- biome lint --write . --unsafe",
 "type-check": "dotenv -e .env.local -- biome check --staged --write . --unsafe --no-errors-on-unmatched",
```
However, these scripts are not suitable for CI checks as they use the `--write` flag, which is used to auto-fix lint or check issues. In CI, it would just fix those issues in the checkout repo and do nothing. We would ideally want it to check for issues and throw errors. Therefore, I have added new scripts in the `package.json` file and used them in the workflow.

I am almost done with the changes and will raise the PR soon. I am not sure about the script names that I have added. Please provide suggestions for those in the PR or any other changes.
